multicluster/controllers/multicluster/common/helper.go

```diff
@@ -52,3 +54,21 @@ func RemoveString(slice []string, s string) (result []string) {
 	}
 	return
 }
+
+// FilterSubsets keeps IPs only and remove others which are unnecessary information for other member cluster.
+func FilterSubsets(subsets []corev1.EndpointSubset) []corev1.EndpointSubset {
+	newSubsets := []corev1.EndpointSubset{}
+	for _, s := range subsets {
+		subset := corev1.EndpointSubset{}
+		newAddresses := []corev1.EndpointAddress{}
+		for _, addr := range s.Addresses {
+			newAddresses = append(newAddresses, corev1.EndpointAddress{
+				IP: addr.IP,
+			})
+		}
+		subset.Addresses = newAddresses
+		subset.Ports = s.Ports
+		newSubsets = append(newSubsets, subset)
```

Comment:
It is not possible subset.Addresses is empty here?

Reply:
yeah, I think it's possible, then newSubset will also be empty. in final MC Endpoint, it might be <None> if no available Endpoints at all.

Reply:
I meant if  subset.Addresses  is empty should we still append it to newSubsets?

Reply:
ah, yes, you are correct, I added a step to check the length before adding the subset.